src/game/AI/ScriptDevAI/scripts/kalimdor/dustwallow_marsh.cpp

```diff
@@ -1498,6 +1511,462 @@ UnitAI* GetAI_npc_smolderwing(Creature* pCreature)
     return new npc_smolderwing(pCreature);
 }
 
+/*######
+## npc_theramore_combatant
+######*/
+
+enum {
+    NPC_GUARD_NARRISHA = 5093,
+    NPC_GUARD_KAHIL = 5091,
+    NPC_GUARD_EDWARD = 4922,
+    NPC_GUARD_TARK = 5094,
+    NPC_GUARD_LANA = 5092,
+    NPC_GUARD_JARAD = 4923,
+
+    NPC_MEDIC_HELAINA = 5200, //Red healer
+    NPC_MEDIC_TAMBERLYN = 5199, //Blue healer
+
+    NPC_MASTER_SZIGETI = 5090,
+    NPC_MASTER_CRITON = 4924,
+
+    NPC_BLUE_CAPTAIN = 5096,
+    NPC_RED_CAPTAIN = 5095,
+
+    SPELL_FIRST_AID = 7162,
+
+    BLUE_TEAM = 1621,
+    RED_TEAM = 1622,
+
+    FACTION_THERAMORE_FRIENDLY = 151,
+
+    SAY_RED_WINNER = 1794, //Szigeti
+    SAY_BLUE_WINNER = 1795, //Szigeti
+    SAY_RED_WINNER_OVERALL = 1773, //Criton
+    SAY_BLUE_WINNER_OVERALL = 1774, //Criton
+
+    SAY_BACK_TO_CORNERS = 1636, //Criton
+
+    SAY_FOR_THE_RED = 1778,
+    SAY_FOR_THE_BLUE = 1777,
+
+    SAY_FIRST_COMBATANTS = 1771,
+    SAY_FIGHT_INTRO = 1643,
+    SAY_FIGHT_START = 1641,
+
+    SAY_EDWARD_LOSS = 1760,
+    SAY_KAHIL_LOSS = 1762,
+    SAY_NARRISHA_LOSS = 1763,
+
+    SAY_JARAD_LOSS = 1766,
+    SAY_LANA_LOSS = 1767,
+    SAY_TARK_LOSS = 1768,
+
+    SAY_RED_MOTIVATION_WIN = 1769,
+    SAY_RED_MOTIVATION_LOSS = 1770,
+
+    SAY_BLUE_MOTIVATION_LOSS = 1765,
+    SAY_BLUE_MOTIVATION_WIN = 1764,
+
+    SAY_RED_FIRST = 1776,
+    SAY_BLUE_FIRST = 1775,
+};
+
+enum CombatantActions
+{
+    COMBATANT_CHECK_HP,
+    COMBATANT_BLUE_WIN,
+    COMBATANT_RED_WIN,
+    COMBATANT_START_POSITION,
+    COMBATANT_START_SPAR,
+    COMBATANT_GET_ALL,
+    COMBATANT_SALUTE,
+    COMBATANT_SWITCH,
+    COMBATANT_SEPARATE,
+    COMBATANT_INTRO,
+    COMBATANT_MOTIVATION,
+    COMBATANT_ACTION_MAX,
+};
+
+static std::unordered_map<uint32, const Position> startingPositions ({
+    {BLUE_TEAM, {-3643.574707, -4505.717773, 9.462863, 3.855592}},
+    {RED_TEAM, {-3652.405273, -4514.126953, 9.462863, 0.717925}}
+});
+
+static const uint32 theramore_blue_team[] = {NPC_GUARD_EDWARD, NPC_GUARD_KAHIL, NPC_GUARD_NARRISHA};
+static const uint32 theramore_red_team[] = {NPC_GUARD_JARAD, NPC_GUARD_LANA, NPC_GUARD_TARK};
+
+struct npc_theramore_spar_controller : public CombatAI
+{
+    npc_theramore_spar_controller(Creature* creature) : CombatAI(creature, COMBATANT_ACTION_MAX)
+    {
+        AddCustomAction(COMBATANT_CHECK_HP, true, [&](){
```

Comment:
These calls can be turned into a slew of oneliners, however if that is not up to your fancy, just newline the {